### PR TITLE
Changed grouping so that same agent can host multiple groups

### PR DESCRIPTION
**Pull request by mullvaden**

### CODE REVIEW

File: src/AsimovDeploy.WinAgent.Tests/ConfigurationSpecs.cs [hunk 1]
```diff
@@ -148,7 +148,25 @@ public void can_match_agent_names_with_regex_range()
             queryString6.ShouldBe("partition=testgroup2&host=a+host");
         }
 
-        public AsimovConfig ReadConfig(string configDir, string agentName)
+		[Test]
```
Comment (bergquist) [inline src/AsimovDeploy.WinAgent.Tests/ConfigurationSpecs.cs]: Gräslig formatering.

**Comment by marefr** [comment]
Really hard to review the actual feature when you've updated the .NET version for all the projects and introduced some C# 6 features. I will accept it this time, but please have in mind that next time I would suggest to create a separate pull request for bigger "structural/refactoring" changes like upgrading .NET versions etc.

I will accept this pull request as soon you add an integration test for this new feature, possibly extending the BasicScenario tests?

**Comment by mullvaden** [comment]
But there is some new tests for this where we deal with this new feature.

(Patrik has given me access now so now I'm able to accept my own PRs..;-)

But yes, the C# might have been better of in another PR.

**Comment by marefr** [comment]
As I said earlier today I've seen your unit tests and they're great, but would be good with an integration test as well.

**Comment by mullvaden** [comment]
Ping Marcus!

**Pull request merged**